demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+ */
+#define USER_BUFFER_LENGTH                ( 1024 )
+
+/**
+ * @brief Length of an IPv6 address when converted to hex digits.
+ */
+#define IPV6_LENGTH                       ( 40 )
+
+
+/**
+ * @brief Some text to send as the request body for PUT and POST requests in
+ * this demo.
+ */
+#define REQUEST_BODY_TEXT           "Hello, world!"
+#define REQUEST_BODY_TEXT_LENGTH    ( sizeof( REQUEST_BODY_TEXT ) - 1 )
+
+/**
+ * @brief A string to store the resolved IP address from the host name.
+ */
+static char resolvedIpAddr[ IPV6_LENGTH ] = { 0 };
+
+/**
+ * @brief A buffer used to store request headers and reused after sending
+ * the request to store the response headers and body.
+ *
+ * User can also decide to have two separate buffers for request and response.
+ */
+static uint8_t userBuffer[ USER_BUFFER_LENGTH ] = { 0 };
+
+/**
+ * @brief The request body.
+ */
+static uint8_t requestBodyBuffer[ REQUEST_BODY_TEXT_LENGTH ] = { 0 };
+
+/**
+ * @brief Definition of the HTTP network context.
+ */
+struct HTTPNetworkContext
+{
+    int tcpSocket;
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Performs a DNS lookup on the given host name then establishes a TCP
+ * connection to the server.
+ *
+ * @param[in] pServer Host name of server.
+ * @param[in] port Server port.
+ * @param[out] pTcpSocket Pointer to TCP socket file descriptor.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int connectToServer( const char * pServer,
+                            uint16_t port,
+                            int * pTcpSocket );
+
+/**
+ * @brief The transport send function that defines the transport interface.
+ *
+ * This is passed to the #HTTPTransportInterface.send function and used to
+ * send data over the network.
+ *
+ * @param[in] pContext User defined context (TCP socket for this demo).
+ * @param[in] pBuffer Buffer containing the bytes to send over the network stack.
+ * @param[in] bytesToSend Number of bytes to write to the network.
+ *
+ * @return Number of bytes sent; negative value on error.
+ */
+static int32_t transportSend( HTTPNetworkContext_t * pContext,
+                              const void * pBuffer,
+                              size_t bytesToSend );
+
+/**
+ * @brief The transport receive function that defines the transport interface.
+ *
+ * @param[in] pContext User defined context (TCP socket for this demo).
+ * @param[out] pBuffer Buffer to read network data into.
+ * @param[in] bytesToRead Number of bytes requested from the network.
+ *
+ * This is passed to the #HTTPTransportInterface.recv function and used to
+ * receive data over the network.
+ *
+ * @return Number of bytes received; negative value on error.
+ */
+static int32_t transportRecv( HTTPNetworkContext_t * pContext,
+                              void * pBuffer,
+                              size_t bytesToRecv );
+
+/**
+ * @brief Send an HTTP request based on a specified method and path.
```

Comment:
Here we must specify that this function will also process the response because API does process the response.